### PR TITLE
[1.11] setup/settings: Update to allow for validationServerGrpcMaxSizeBytes to take effect without manual kick

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

On settings update we check if grpc servers need a kick
Right now we only do it if the addr changes but we also need to set anything else that is passed in as a grpc opt. In this case the validation bytes.

# Context

Users were confused when the settings change was not reflected in the validation server's behavior

### CODE REVIEW

File: projects/gloo/pkg/syncer/setup/setup_syncer.go [hunk 1]
```diff
@@ -112,8 +112,9 @@ func NewSetupFuncWithRunAndExtensions(runFunc RunFunc, extensions *Extensions) s
 }
 
 type grpcServer struct {
-	addr   string
-	cancel context.CancelFunc
+	addr            string
+	maxGrpcRecvSize int
```
Comment (jackstine) [inline projects/gloo/pkg/syncer/setup/setup_syncer.go:116]: nit: comment about what `maxGrpcRecvSize` is doing.

File: projects/gloo/pkg/syncer/setup/setup_syncer.go [hunk 2]
```diff
@@ -225,7 +238,8 @@ func (s *setupSyncer) Setup(ctx context.Context, kubeCache kube.SharedCache, mem
 		s.controlPlane = emptyControlPlane
 	}
 
-	if validationAddr != s.previousValidationServer.addr {
+	// check if we need to restart the validation server
```
Comment (jackstine) [inline projects/gloo/pkg/syncer/setup/setup_syncer.go:241]: Nice comments

File: projects/gloo/pkg/syncer/setup/setup_syncer.go (hunk 1 above)
Comment (jbohanon) [reply projects/gloo/pkg/syncer/setup/setup_syncer.go:118]: added https://github.com/solo-io/gloo/pull/6608/commits/f2004485e02a0bb0468f1dab4b7e73de27168d64

File: test/kube2e/gloo/setup_syncer_test.go [hunk 3]
```diff
@@ -142,6 +149,50 @@ var _ = Describe("SetupSyncer", func() {
...
+				Expect(err).NotTo(HaveOccurred())
+
+				newContext()
+				settings.Gateway.Validation.ValidationServerGrpcMaxSizeBytes = &wrappers.Int32Value{Value: 1}
+				err = setup(ctx, nil, memcache, settings)
+				Expect(err).NotTo(HaveOccurred())
+
+				// make sure that validation server was restarted with new value
+				// it should now reject the request
+				Eventually(func() string {
```
Comment (jackstine) [inline test/kube2e/gloo/setup_syncer_test.go:188]: nit: maybe reword the comments here to make it clear that the goal of the test is to test that resizing `ValidationServerGrpcMaxSizeBytes` will cause an error and confirm that the max size has changed.

Comment (jbohanon) [reply test/kube2e/gloo/setup_syncer_test.go:189]: revised https://github.com/solo-io/gloo/pull/6608/commits/1b1e9d73b9ab725327d840102b4d1c51ae42bc9e

**Review by jackstine — APPROVED** [review]
LGTM!
